actors/states/check.go

```diff
@@ -213,3 +214,40 @@ func CheckMinersAgainstPower(acc *builtin.MessageAccumulator, minerSummaries map
 		acc.Require(provingPeriodCron != nil, "miner %v has no proving period cron", addr)
 	}
 }
+
+func CheckDealStatesAgainstSectors(acc *builtin.MessageAccumulator, minerSummaries map[addr.Address]*miner.StateSummary, marketSummary *market.StateSummary) {
+	for dealID, deal := range marketSummary.Deals { // nolint:nomaprange
+		if deal.SectorStartEpoch == abi.ChainEpoch(-1) {
+			// deal hasn't been activated yet, make no assertions about sector state
+			continue
+		}
+
+		minerSummary, found := minerSummaries[deal.Provider]
+		if !found {
+			acc.Addf("provider %v for deal %d not found among miners", deal.Provider, dealID)
+			continue
+		}
+
+		sectorDeal, found := minerSummary.Deals[dealID]
+		if !found {
+			acc.Addf("deal %d not referenced in active sectors of miner %v", dealID, deal.Provider)
+			continue
+		}
+
+		acc.Require(deal.SectorStartEpoch == sectorDeal.SectorStart,
+			"deal state start %d does not match sector start %d for miner %v",
+			deal.SectorStartEpoch, sectorDeal.SectorStart, deal.Provider)
+
+		acc.Require(deal.SectorStartEpoch <= sectorDeal.SectorExpiration,
+			"deal state start %d activated after sector expiration %d for miner %v",
+			deal.SectorStartEpoch, sectorDeal.SectorExpiration, deal.Provider)
+
+		acc.Require(deal.LastUpdatedEpoch <= sectorDeal.SectorExpiration,
+			"deal state update at %d after sector expiration %d for miner %v",
+			deal.LastUpdatedEpoch, sectorDeal.SectorExpiration, deal.Provider)
+
+		acc.Require(deal.SlashEpoch <= sectorDeal.SectorExpiration,
+			"deal state slashed at %d after sector expiration %d for miner %v",
+			deal.SlashEpoch, sectorDeal.SectorExpiration, deal.Provider)
+	}
```

Comment:
I was going to ask to check that all sector deals are in market state, but I realise some of them might have expired and been removed. Leave a comment there explaining why we can't check that.

Reply:
I added a comment above the loop.